CRM/ManualDirectDebit/Form/Batch.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+use CRM_ManualDirectDebit_ExtensionUtil as E;
+
+/**
+ * This class generates form components for Create Instructions Batch
+ */
+class CRM_ManualDirectDebit_Form_Batch extends CRM_Admin_Form {
+
+  /**
+   * PreProcess function.
+   *
+   */
+  public function preProcess() {
+    parent::preProcess();
+
+    $batchTypeID = CRM_Utils_Request::retrieveValue('type_id', 'String', NULL);
+    $batchType = CRM_Core_OptionGroup::getRowValues('batch_type', $batchTypeID, 'value', 'String', FALSE);
+
+    // Set the user context.
+    $session = CRM_Core_Session::singleton();
+    $session->replaceUserContext(CRM_Utils_System::url('civicrm/direct_debit/batch', "reset=1&action=add&type_id=" . $batchType['value']));
+
+    CRM_Utils_System::setTitle(E::ts('Create %1', [1 => $batchType['label']]));
+
+    $this->assign('batch_id', CRM_ManualDirectDebit_Batch_BatchHandler::getMaxBatchId() + 1);
+    $this->add('hidden', 'batch_id', CRM_ManualDirectDebit_Batch_BatchHandler::getMaxBatchId() + 1);
```

Comment:
Just realized we could have problems too here, although unlikely. I'd still feel more comfortable if we did:
```php
$newBatchID = CRM_ManualDirectDebit_Batch_BatchHandler::getMaxBatchId() + 1;
$this->assign('batch_id', $newBatchID);
$this->add('hidden', 'batch_id', $newBatchID);
```